### PR TITLE
Edl nginx

### PR DESCRIPTION
**Pull request by glicht**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Use nginx with EDL

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 1]
```diff
@@ -486,22 +530,37 @@ def route_edl_values() -> Response:
...
         on_demand=on_demand,
         request_args=request_args,
-        cache_refresh_rate=params.get('cache_refresh_rate'),
+        cache_refresh_rate=cache_refresh_rate,
     )
-    return Response(values, status=200, mimetype='text/plain')
+    query_time = (datetime.now(timezone.utc) - created).total_seconds()
+    edl_size = values.count('\n') + 1  # add 1 as last line doesn't have a \n
+    max_age = ceil((datetime.now() - dateparser.parse(cache_refresh_rate)).total_seconds())
+    demisto.debug(f'Returning edl of size: [{edl_size}], created: [{created}], query time seconds: [{query_time}],'
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.py:554]: @glicht `demisto.debug` is not thread-safe. I'm wondering if it should be protected with a lock. This comment should also be true for the handlers using `demisto.debug` and `demisto.info`.

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 2]
```diff
@@ -576,63 +635,127 @@ def test_module(_: Dict, params: Dict):
...
+        ssl = 'ssl'  # to be included in the listen directive
+        sslcerts = NGINX_SSL_CERTS
+    credentials = params.get('credentials') or {}
+    if credentials.get('identifier'):
+        extra_cache_key = "$http_authorization"
+    server_conf = Template(template_str).safe_substitute(port=port, serverport=serverport, ssl=ssl,
+                                                         sslcerts=sslcerts, extra_cache_key=extra_cache_key)
+    with open(file_path, mode='wt') as f:
+        f.write(server_conf)
+    return port
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.py:673]: per our discussion, no need to return port.

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 3]
```diff
@@ -576,63 +635,127 @@ def test_module(_: Dict, params: Dict):
...
+        APP.logger.addHandler(log_handler)
+        demisto.debug('done setting demisto handler for logging')
+        server = WSGIServer(('0.0.0.0', server_port), APP, log=DEMISTO_LOGGER, error_log=ERROR_LOGGER)
         if is_test:
+            test_nginx_server(nginx_port, params)
             server_process = Process(target=server.serve_forever)
             server_process.start()
             time.sleep(5)
-            server_process.terminate()
+            server_process.terminate()
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.py:746]:
```suggestion
            server_process.terminate()
```

**Review by DeanArbel — APPROVED** [review]
Impressive changes. I have some small comments.

File: Packs/EDL/Integrations/EDL/EDL.py (hunk 1 above)
Comment (glicht) [reply Packs/EDL/Integrations/EDL/EDL.py:560]: It's ok. As we use gevent we are single threaded.

**Comment by glicht** [comment]
@DeanArbel also added monitoring to nginx logs. See: `nginx_log_monitor_loop`.

**Comment by glicht** [comment]
@amshamah419 please help with the release notes review. Also a few changes in the param descriptions.

**Comment by amshamah419** [comment]
Docs look good.

**Pull request merged**